rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,175 @@
+        returns sample from the Tulap distribution.
+    \end{lemma}
+    \end{note}
+\end{tcolorbox}
+
+\texttt{pinpoint} can only fail when the pseudorandom byte generator used in its implementation fails due to lack of system entropy. 
+This is usually related to the computer's physical environment and not the dataset. 
+The rest of this proof is conditioned on the assumption that \texttt{pinpoint} does not raise an exception. \\
+
+A canonical noise distribution (CND)\parencite{awan2023canonical}, 
+captures whether a real-valued distribution is perfectly tailored to satisfy $f$-DP\parencite{dong2019gaussian}. 
+We formalize this in Definition \ref{def2}\parencite{awan2023canonical}. 
+\begin{defn}\label{def2}  % Def 3.1
+Let $f$ be a symmetric nontrivial tradeoff function. A {continuous} distribution function $F$ is a \emph{canonical noise distribution} (CND) for $f$ if 
+\begin{enumerate}[(1)]
+    % Mike: there was an \mscr after "S:". What is it for?
+    \item for every statistic $S: X^n\rightarrow \mathbb{R}$ with sensitivity $\Delta>0$, and $N\sim F(\cdot)$, 
+        the mechanism $S(X) + \Delta N$ satisfies $f$-DP. Equivalently, for every $m\in [0,1]$, $T(F(\cdot),F(\cdot-m))\geq f$,
+    \item $f(\alpha)=T(F(\cdot),F(\cdot-1))(\alpha)$ for all $\alpha \in (0,1)$,
+    \item $T(F(\cdot),F(\cdot-1))(\alpha) = F(F^{-1}(1-\alpha)-1)$ for all $\alpha \in (0,1)$,
+    \item $F(x) = 1-F(-x)$ for all $x\in \mathbb{R}$; that is, $F$ is the cdf of a random variable which is symmetric about zero.
+\end{enumerate}
+\end{defn}
+
+\begin{defn} \label{def1}  % Def 3.7
+Let $f$ be a symmetric nontrivial tradeoff function, and let {$c\in [0,1/2)$} be the unique fixed point of $f$: $f(c)=c$. 
+We define $F_f:\mathbb{R}\rightarrow \mathbb{R}$ as  \[ F_f(x) = \begin{cases}
+    f(1-F_f(x+1))&x<-1/2\\
+    c(1/2-x) + (1-c)(x+1/2)&-1/2\leq x\leq 1/2\\
+    1-f(F_f(x-1))&x>1/2.\\
+\end{cases}\]
+\end{defn}
+In Definition \ref{def1} \parencite{awan2023canonical}, 
+the fact that there is a unique fixed point follows from the fact that $f$ is convex and decreasing, 
+and so intersects the line $y=\alpha$ at a unique value. 
+Note that in Definition \ref{def1} \parencite{awan2023canonical}, 
+the cumulative distribution function (CDF) corresponds to a uniform random variable on the interval $[-1/2,1/2]$. 
+However, due to the recursive nature of $F_f$ and the fact that $f$ is generally non-linear, 
+the Canonical Noise Distribution (CND) from Definition \ref{def1} \parencite{awan2023canonical} need not be uniformly distributed on any other intervals.
+
+Theorem \ref{thm1} \parencite{awan2023canonical} below states that for any nontrivial tradeoff function, 
+this construction yields a CND, which can be constructed as in Definition \ref{def1} \parencite{awan2023canonical}. 
+This CND can be used to add perfectly calibrated noise to a statistic to achieve $f$-DP \parencite{dong2019gaussian}. 
+
+\begin{thm} \label{thm1}  % Thm 3.9
```

Comment:
Is this a theorem you are proving or are you citing it from somewhere else? If it's the former, clearly indicate where the proof is in a proof environment. If it's the latter, cite where you're taking this from at the beginning of the theorem in the standard way.

Reply:
I would just cite at the start of the theorem environment itself where the theorem is from.

\begin[\cite[Theorem~3.9]{awan2023canonical}]{thm}

Or something like this. It would be much more clear this way.